pkg/service/service.go

```diff
@@ -729,6 +744,72 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 	return new, lb.ID(svc.frontend.ID), nil
 }
 
+// upsertNodePortHealthService makes the HealthCheckNodePort available to the external IP of the service
+func (s *Service) upsertNodePortHealthService(svc *svcInfo) error {
```

Comment:
One approach to adding a test here would be to parameterize this function with a new interface, defined in this file, similar to the following

```
type NodeMetaCollector interface {
	GetIPv4() net.IP
	GetIPv6() net.IP
}

type nodeMetaCollector struct{}

func (n *nodeMetaCollector) GetIPv4() net.IP {
	return node.GetIPv4()
}

func (n *nodeMetaCollector) GetIPv6() net.IP {
	return node.GetIPv6()
}
```

A test could then provide an alternate implementation for `NodeMetaCollector`

Ideally, this package, and the node package, would be refactored into a `hive.Cell` in order to allow dependency injection of `node`. However, that's not necessary for the purposes of this PR.

Reply:
Ty for showing the way @asauber, I have created the tests that @joamaki suggested